flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptorFactory.java

```diff
@@ -156,20 +141,32 @@ private List<InputGateDeploymentDescriptor> createInputGateDeploymentDescriptors
                             subpartitionRange,
                             consumedPartitionGroup.size(),
                             getConsumedPartitionShuffleDescriptors(
-                                    consumedIntermediateResult, consumedPartitionGroup)));
+                                    consumedIntermediateResult,
+                                    consumedPartitionGroup,
+                                    executionVertex.getExecutionGraphAccessor())));
         }
 
-        for (Map.Entry<IntermediateDataSetID, ShuffleDescriptorAndIndex[]> entry :
-                consumedClusterPartitionShuffleDescriptors.entrySet()) {
-            // For FLIP-205, the JobGraph generating side ensure that the cluster partition is
-            // produced with only one subpartition. Therefore, we always consume the partition with
-            // subpartition index of 0.
-            inputGates.add(
-                    new InputGateDeploymentDescriptor(
-                            entry.getKey(),
-                            ResultPartitionType.BLOCKING_PERSISTENT,
-                            0,
-                            entry.getValue()));
+        try {
+            for (Map.Entry<IntermediateDataSetID, ShuffleDescriptorAndIndex[]> entry :
```

Comment:
I prefer to keep the same logic as before, only put `getClusterPartitionShuffleDescriptors` in try catch block.